index.js

```diff
@@ -53,14 +53,24 @@ function ensureAllCommentsPrinted(astComments) {
   });
 }
 
-function format(text, opts, addAlignmentSize) {
+function formatWithCursor(text, opts, addAlignmentSize) {
   addAlignmentSize = addAlignmentSize || 0;
 
   const ast = parser.parse(text, opts);
 
   const formattedRangeOnly = formatRange(text, opts, ast);
   if (formattedRangeOnly) {
-    return formattedRangeOnly;
+    return { formatted: formattedRangeOnly };
+  }
+
+  let cursorOffset;
+  if (opts.cursorOffset >= 0) {
+    const cursorNodeAndParents = findNodeAtOffset(ast, opts.cursorOffset);
+    const cursorNode = cursorNodeAndParents.node;
+    if (cursorNode) {
+      cursorOffset = opts.cursorOffset - util.locStart(cursorNode);
+      opts.cursorNode = cursorNode;
```

Comment:
Hmm, we're already mutating elsewhere with `opts.newLine = guessLineEnding(text);` and `opts.originalText = text.trimRight();`. I can take a look refactoring it tomorrow though, if you're set on this.